amqp/src/main/java/org/axonframework/amqp/eventhandling/DefaultAMQPMessageConverter.java

```diff
@@ -112,21 +114,87 @@ public AMQPMessage createAMQPMessage(EventMessage<?> eventMessage) {
+                                                               (Long) headers.get(Headers.AGGREGATE_SEQ),
+                                                               message, () -> DateTimeUtils.parseInstant(timestamp)));
         } else {
             return Optional.of(new GenericEventMessage<>(message, () -> DateTimeUtils.parseInstant(timestamp)));
         }
     }
 
+    /**
+     * Builder class to instantiate a {@link DefaultAMQPMessageConverter}.
+     * The {@link RoutingKeyResolver} is defaulted to a {@link PackageRoutingKeyResolver} and the {@code durable} field
+     * defaults to {@code true}. The {@link Serializer} is a <b>hard requirement</b> and thus should be provided.
+     */
+    public static class Builder {
+
+        private Serializer serializer;
+        private RoutingKeyResolver routingKeyResolver = new PackageRoutingKeyResolver();
+        private boolean durable = true;
+
+        /**
+         * Sets the serializer to serialize the Event Message's payload and Meta Data with.
+         *
+         * @param serializer The serializer to serialize the Event Message's payload and Meta Data with
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder serializer(Serializer serializer) {
+            assertNonNull(serializer, "Serializer may not be null");
+            this.serializer = serializer;
+            return this;
+        }
+
+        /**
+         * Sets the strategy to use to resolve routing keys for Event Messages. Defaults to a
+         * {@link PackageRoutingKeyResolver}.
+         *
+         * @param routingKeyResolver The strategy to use to resolve routing keys for Event Messages
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder routingKeyResolver(RoutingKeyResolver routingKeyResolver) {
+            assertNonNull(routingKeyResolver, "RoutingKeyResolver may not be null");
+            this.routingKeyResolver = routingKeyResolver;
+            return this;
+        }
+
+        /**
+         * Sets a {@code boolean} specifying whether to request durable message dispatching. Defaults to {@code true},
+         * thus toggling durable message dispatching on.
+         *
+         * @param durable a {@code boolean} specifying whether to request durable message dispatching
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder durable(boolean durable) {
+            this.durable = durable;
+            return this;
+        }
+
+        /**
+         * Initializes a {@link DefaultAMQPMessageConverter} as specified through this Builder.
+         *
+         * @return a {@link DefaultAMQPMessageConverter} as specified through this Builder
+         */
+        public DefaultAMQPMessageConverter build() {
+            return new DefaultAMQPMessageConverter(this);
+        }
+
+        private void validate() {
```

Comment:
One minor thing, I would make `validate` methods `protected` due to inheritance.

Reply:
I figured this was irrelevant, as the `validate()` function of every Builder will be called eventually.
Implementations of the `DefaultAMQPMessageConverter` will *have to* call the super constructor.
And the only super constructor present already calls the `validate()` function.

Hence, no need to make them `protected`, imo.
If I would override this function, thus calling `super.validate()` internally added with the new assertions, we'd be duplicating a lot of function calls.